Automattic/jetpack
# PHPCS: Add `use` statement sorting sniff

#### Changes proposed in this Pull Request:

After a merge of two patches adding `use` statements for the same class in different places in the existing list broke things, discussion on Slack was in favor of adding a sniff to require that `use` statements are in alphabetical order.

As https://github.com/wikimedia/mediawiki-tools-codesniffer already has just such a sniff, we'll copy theirs and modify it to suit our purposes.

This also adjusts .phpcs.xml.dist to ignore more files that aren't actually part of the repo, in preparation for the followup (#17362) that mass-fixes all the new sniff failures this patch introduces.

#### Jetpack product discussion


p9dueE-1Y9-p2 (and some earlier Slack discussion)

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Edit a file to have an unsorted `use` list.
* `composer php:lint file.php` should detect it.
* `composer php:autofix file.php` should fix it.

#### Proposed changelog entry for your changes:

* Developer-only, no changelog entry needed.
